code/Language/Drasil/Sentence/Extract.hs

```diff
@@ -0,0 +1,46 @@
+module Language.Drasil.Sentence.Extract(sdep, vars',snames, combine) where
+
+import Data.List (nub)
+import Control.Lens ((^.))
+import Language.Drasil.Expr (Expr(..), RealInterval(..))
+import Language.Drasil.ChunkDB (HasSymbolTable, symbLookup, symbolTable, HasDefinitionTable,
+ defLookup, defTable)
+import Language.Drasil.Chunk.Code (CodeChunk, codevar)
+import Language.Drasil.Chunk.Quantity (QuantityDict)
+import Language.Drasil.Chunk.Concept (ConceptChunk)
+import Language.Drasil.Chunk.DefinedQuantity
+import Language.Drasil.Spec(Sentence(..))
+import Language.Drasil.Expr.Extract(names)
+
+-- | Generic traverse of all positions that could lead to names from sentences
+snames   :: Sentence -> [String]
+snames (Ch a)       = [a]
+snames (Sy _)       = []
+snames (S _)        = []
+snames (Sp _)       = []
+snames (P _)        = []
+snames (Ref _ _ _)  = []
+snames ((:+:) a b)  = (snames a) ++ (snames b)
+snames (Quote a)    = snames a
+snames (E a)        = names a
+snames (EmptyS)     = []
+-----------------------------------------------------------------------------
+-- And now implement the exported traversals all in terms of the above
+sdep :: Sentence -> [String]
+sdep = nub . snames
+
+vars' :: (HasSymbolTable s) => Sentence -> s -> [QuantityDict]
```

Comment:
I believe that these 3 functions (`vars'`, `concpt` and `combine`) should probably be defined elsewhere. Likely closer to where they are used.

Reply:
I created a new module called "ChunkBack" in drasil-lang.Language.Drasil.Document.ChunkBack to put constructors who can use different uids to get back to different type chunks. I am not sure whether this idea is proper. Changes were made in c048939df6606a7dd66380a03c37177868d39b99. (This commit built successfully on local machine but fail on remote repo, trying to figure out why but logic should be fine).

Reply:
You probably saw already that I commented on the name - but otherwise the idea is good. This way the module dependency graph won't be too bad.

Did you maybe forget to add the new file to the repo? [I am asking because that's one thing that happens to me once in a while]

Reply:
I have fixed the error. I left some files that I modified open but didn't save them.